### PR TITLE
[SYCLomatic] Add option to print path to helper function

### PR DESCRIPTION
**Pull request by daiyaan-ahmed6**
Print path to installed directory of helper functions.

Add --helper-function to autocomplete.c LIT test.

Changes made in getInstalledPath() and ValidateInputDirectory() to remove unused parameter 'Tool'.

Signed-off-by: Ahmed, Daiyaan <daiyaan.ahmed@intel.com>

### CODE REVIEW

File: clang/lib/DPCT/DPCT.cpp [hunk 1]
```diff
@@ -603,6 +601,23 @@ int runDPCT(int argc, const char **argv) {
   }
 
   initWarningIDs();
+
+  DpctInstallPath = getInstallPath(argv[0]);
+
+  if (PathToHelperFunction) {
+    SmallString<512> pathToHelperFunction(DpctInstallPath);
```
Comment (zhimingwang36) [inline clang/lib/DPCT/DPCT.cpp:604]: Install path include "bin",  so please remove the "bin"

File: clang/test/dpct/autocomplete.c [hunk 2]
```diff
@@ -24,6 +24,7 @@
 // DASH-NEXT: --gen-build-script
```
Comment (zhimingwang36) [inline clang/test/dpct/autocomplete.c:24]: Please add lit test for the new added option

File: clang/include/clang/DPCT/DPCTOptions.inc [hunk 3]
```diff
@@ -414,6 +414,10 @@ DPCT_NON_ENUM_OPTION(DPCT_OPT_TYPE(static llvm::cl::opt<bool>), GenHelperFunctio
         llvm::cl::desc("Generate helper function files in the --out-root directory. Default: off."),
         llvm::cl::cat(DPCTCat), llvm::cl::init(false))
 
+DPCT_NON_ENUM_OPTION(DPCT_OPT_TYPE(static llvm::cl::opt<bool>), PathToHelperFunction, "helper-function",
+        llvm::cl::desc("Prints the path to helper functions"),
```
Comment (zhimingwang36) [inline clang/include/clang/DPCT/DPCTOptions.inc]: ### 
```suggestion
        llvm::cl::desc("Print the installation directory for helper function header files."),
```

File: clang/include/clang/DPCT/DPCTOptions.inc [hunk 4]
```diff
@@ -414,6 +414,10 @@ DPCT_NON_ENUM_OPTION(DPCT_OPT_TYPE(static llvm::cl::opt<bool>), GenHelperFunctio
         llvm::cl::desc("Generate helper function files in the --out-root directory. Default: off."),
         llvm::cl::cat(DPCTCat), llvm::cl::init(false))
 
+DPCT_NON_ENUM_OPTION(DPCT_OPT_TYPE(static llvm::cl::opt<bool>), PathToHelperFunction, "helper-function",
```
Comment (zhimingwang36) [inline clang/include/clang/DPCT/DPCTOptions.inc]:
```suggestion
DPCT_NON_ENUM_OPTION(DPCT_OPT_TYPE(static llvm::cl::opt<bool>), PathToHelperFunction, "helper-function-dir",
```

File: clang/lib/DPCT/DPCT.cpp (hunk 1 above)
Comment (daiyaan-ahmed6) [reply clang/lib/DPCT/DPCT.cpp:604]: Bin is not included in the DPCTInstallPath variable. Checked.

File: clang/test/dpct/autocomplete.c (hunk 2 above)
Comment (daiyaan-ahmed6) [reply clang/test/dpct/autocomplete.c:24]: LIT test for --helper-function-dir line added to autocomplete.c 

E2E test for this feature added here: https://github.com/oneapi-src/SYCLomatic-test/pull/438

File: clang/include/clang/DPCT/DPCTOptions.inc (hunk 3 above)
Comment (daiyaan-ahmed6) [reply clang/include/clang/DPCT/DPCTOptions.inc]: Done.

**Review by tangjj11 — APPROVED** [review]
lgtm

File: clang/include/clang/DPCT/DPCTOptions.inc (hunk 4 above)
Comment (daiyaan-ahmed6) [reply clang/include/clang/DPCT/DPCTOptions.inc]: Done

**Review by tomflinda — APPROVED** [review]
LGTM

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**